### PR TITLE
We should state in setup.py that astropy is python 3.5+ only

### PR DESCRIPTION
**Pull request by bsipocz**
To make sure v3.0 is not installable on python2.7

### CODE REVIEW

**Comment by eteq** [comment]
(Note that this is a follow-on to #6556, which removed all the py 2.x support)

File: astropy/__init__.py [hunk 1]
```diff
@@ -11,8 +11,11 @@
 import os
 from warnings import warn
 
-if sys.version_info[:2] < (3, 5):
-    warn("Astropy does not support Python <3.5 (in v3.0 and later)")
+__minimum_python_version__ = '3.5'
+__minimum_numpy_version__ = '1.9.0'
+
+if sys.version_info < tuple((int(val) for val in __minimum_python_version__.split('.'))):
```
Comment (MSeifert04) [inline astropy/__init__.py]: can't we use `LooseVersion` here? (from `distutils` or so).

**Comment by eteq** [comment]
(discussed offline with @bsipocz): turns out this doesn't quite catch the failure in py 2.x because ``astropy/__init__.py`` yields a SyntaxError *before* setup.py reaches the setup function.  There *is* however a warning that the `__init__.py` yields.  I've updated this PR to make that stay in sync with the setup.py machinery, while  @bsipocz is going to make a PR to turn that warning into an error (as that's a bit more controversial than the changes here)

File: astropy/__init__.py (hunk 1 above)
Comment (eteq) [reply astropy/__init__.py]: Hmm... Maybe?  I thought the recommended way with `sys.version_info` was to do the direct comparison, which is why `version_info` is such a weird tuple format... And you can't directly compare ``sys.version_info`` or ``sys.version`` with the distutils version objects.  But that predates `LooseVersion`, so maybe there's a way to get a version object that's the current python?

Comment (MSeifert04) [reply astropy/__init__.py]: You're right. `sys.version` shouldn't be used for extracting the `version`. And `LooseVersion('.'.join(map(str, sys.version_info)))` is also ugly...

Comment (eteq) [reply astropy/__init__.py]: The whole version-comparison machinery is kind of a mess, sadly. I really *want* to use `LooseVersion` or similar for everything but there keeps being weird corner cases and changes to the rules that makes it hard to account for.

**Pull request merged**